compiler/front_end/BUILD

```diff
@@ -118,6 +118,7 @@ py_library(
     name = "test_util",
     testonly = 1,
     srcs = ["test_util.py"],
+    visibility = ["//:__subpackages__"],
```

Comment:
Note to reviewers: If it would be more appropriate to move this to `compiler/util` now that it is being used in both the front-end and the back-end (leaving only `glue` to be used directly from the front-end), I'd be happy to do that in this PR as well.

Reply:
Given what's in `test_util.py`, it does make sense to move it to `compiler/util`.

Reply:
Sounds good! Moved to `compiler/util` in latest commit.